### PR TITLE
Warn user that `conda env remove` removes all contents of the prefix path 

### LINKED ISSUE #12222: Conda deletes files it didn't create
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

I created a new Conda environment in a directory that already had files and subdirectories.

Later, I told Conda to remove the environment.  
Conda started emptying the directory with no furthur output.  
I interrupted (<kbd>Ctrl</kbd>+<kbd>C</kbd>) Conda when it was taking too long. 

Half of the directory was gone by then. 

### Conda Info

```shell
active environment : base
    active env location : C:\Users\swast\miniconda3
            shell level : 1
       user config file : C:\Users\swast\.condarc
 populated config files :
          conda version : 22.11.1
    conda-build version : not installed
         python version : 3.10.4.final.0
       virtual packages : __archspec=1=x86_64
                          __cuda=11.7=0
                          __win=0=0
       base environment : C:\Users\swast\miniconda3  (writable)
      conda av data dir : C:\Users\swast\miniconda3\etc\conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\swast\miniconda3\pkgs
                          C:\Users\swast\.conda\pkgs
                          C:\Users\swast\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\swast\miniconda3\envs
                          C:\Users\swast\.conda\envs
                          C:\Users\swast\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/22.11.1 requests/2.28.1 CPython/3.10.4 Windows/10 Windows/10.0.19045
          administrator : False
             netrc file : None
           offline mode : False
```

### Conda Config

_No response_

### Conda list

```shell
# packages in environment at C:\Users\swast\miniconda3:
#
# Name                    Version                   Build  Channel
asttokens                 2.0.5              pyhd3eb1b0_0    defaults
autopep8                  1.6.0                    pypi_0    pypi
backcall                  0.2.0              pyhd3eb1b0_0    defaults
brotlipy                  0.7.0           py310h2bbff1b_1002    defaults
bzip2                     1.0.8                he774522_0    defaults
ca-certificates           2022.10.11           haa95532_0    defaults
certifi                   2022.12.7       py310haa95532_0    defaults
cffi                      1.15.1          py310h2bbff1b_3    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
colorama                  0.4.6           py310haa95532_0    defaults
conda                     22.11.1         py310haa95532_4    defaults
conda-package-handling    1.9.0           py310h8cc25b3_1    defaults
console_shortcut          0.1.1                         4    defaults
cryptography              38.0.1          py310h21b164f_0    defaults
debugpy                   1.5.1           py310hd77b12b_0    defaults
decorator                 5.1.1              pyhd3eb1b0_0    defaults
entrypoints               0.4             py310haa95532_0    defaults
executing                 0.8.3              pyhd3eb1b0_0    defaults
idna                      3.4             py310haa95532_0    defaults
ipykernel                 6.9.1           py310haa95532_0    defaults
ipython                   8.7.0           py310haa95532_0    defaults
jedi                      0.18.1          py310haa95532_1    defaults
jupyter_client            7.4.8           py310haa95532_0    defaults
jupyter_core              5.1.1           py310haa95532_0    defaults
libffi                    3.4.2                hd77b12b_6    defaults
libsodium                 1.0.18               h62dcd97_0    defaults
matplotlib-inline         0.1.6           py310haa95532_0    defaults
menuinst                  1.4.17          py310h59b6b97_0    defaults
nest-asyncio              1.5.6           py310haa95532_0    defaults
openssl                   1.1.1s               h2bbff1b_0    defaults
parso                     0.8.3              pyhd3eb1b0_0    defaults
pickleshare               0.7.5           pyhd3eb1b0_1003    defaults
pip                       22.3.1          py310haa95532_0    defaults
platformdirs              2.5.2           py310haa95532_0    defaults
pluggy                    1.0.0           py310haa95532_1    defaults
powershell_shortcut       0.0.1                         3    defaults
prompt-toolkit            3.0.36          py310haa95532_0    defaults
pure_eval                 0.2.2              pyhd3eb1b0_0    defaults
pycodestyle               2.8.0                    pypi_0    pypi
pycosat                   0.6.4           py310h2bbff1b_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pygments                  2.11.2             pyhd3eb1b0_0    defaults
pyopenssl                 22.0.0             pyhd3eb1b0_0    defaults
pysocks                   1.7.1           py310haa95532_0    defaults
python                    3.10.4               hbb2ffb3_0    defaults
python-dateutil           2.8.2              pyhd3eb1b0_0    defaults
pywin32                   305             py310h2bbff1b_0    defaults
pyzmq                     23.2.0          py310hd77b12b_0    defaults
requests                  2.28.1          py310haa95532_0    defaults
ruamel.yaml               0.17.21         py310h2bbff1b_0    defaults
ruamel.yaml.clib          0.2.6           py310h2bbff1b_1    defaults
setuptools                65.5.0          py310haa95532_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
sqlite                    3.40.0               h2bbff1b_0    defaults
stack_data                0.2.0              pyhd3eb1b0_0    defaults
tk                        8.6.12               h2bbff1b_0    defaults
toml                      0.10.2                   pypi_0    pypi
toolz                     0.12.0          py310haa95532_0    defaults
tornado                   6.2             py310h2bbff1b_0    defaults
tqdm                      4.64.1          py310haa95532_0    defaults
traitlets                 5.7.1           py310haa95532_0    defaults
tzdata                    2022g                h04d1e81_0    defaults
urllib3                   1.26.13         py310haa95532_0    defaults
vc                        14.2                 h21ff451_1    defaults
vs2015_runtime            14.27.29016          h5e58377_2    defaults
wcwidth                   0.2.5              pyhd3eb1b0_0    defaults
wheel                     0.37.1             pyhd3eb1b0_0    defaults
win_inet_pton             1.1.0           py310haa95532_0    defaults
wincertstore              0.2             py310haa95532_2    defaults
xz                        5.2.8                h8cc25b3_0    defaults
zeromq                    4.3.4                hd77b12b_0    defaults
zlib                      1.2.13               h8cc25b3_0    defaults
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ForgottenProgramme**
### Description
Resolves #12222

This PR adds a Y/N prompt warning the user that `conda env remove  -p <prefix_path>` and `conda remove --all -p <prefix_path>` deletes not only the conda environment at <prefix_path> but all the other contents if present. 

- [x] Add y/n prompt to `conda remove` and `conda env remove`
- [x] update `test_notices_does_not_interrupt_command_on_failure` to use `PathFactoryFixture`

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda/cli/main_remove.py [hunk 1]
```diff
@@ -210,6 +212,12 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
                     )
         if not context.dry_run:
             if not args.keep_env:
+                if not args.json:
+                    confirm_yn(
+                        f"WARNING: If the prefix path: {prefix} contains any contents other than the conda environment, these will also be deleted during this action.\n",
```
Comment (kenodegard) [inline conda/cli/main_remove.py:217]:
```suggestion
                        f"All files found within the environment ({prefix}), including any conda environment configurations and any non-conda files, will be deleted. Do you wish to continue?\n",
```

File: news/13440-add-yn-prompt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Add a Y/N prompt warning users that conda env remove  -p <prefix_path> and conda remove --all -p <prefix_path> deletes not only the conda environment at <prefix_path> but all the other contents if present.
```
Comment (kenodegard) [inline news/13440-add-yn-prompt:3]:
```suggestion
* Add a Y/N prompt warning users that `conda env remove` and `conda remove --all` deletes not only the conda packages but all other files present in the specified environment. (#12222 via #13440)
```

Comment (ForgottenProgramme) [reply news/13440-add-yn-prompt:3]: Hey, @kenodegard !
What we are trying to tell the users in that the entire directory that contains the environment will be deleted. I am thinking of the case when a user creates an environment in a non empty directory and then tries to delete the environment. In this case they should know that the directory in which the environment lives will also be deleted.

**Review by jezdez — APPROVED** [review]
With @kenodegard's comments!

File: news/13440-add-yn-prompt (hunk 2 above)
Comment (kenodegard) [reply news/13440-add-yn-prompt:3]: ok so we modify it to:

```suggestion
* Add a Y/N prompt warning users that `conda env remove` and `conda remove --all` deletes not only the conda packages but all other contents in the specified environment. (#12222 via #13440)
```

**Review by beeankha — DISMISSED** [review]

**Review by kenodegard — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**